src/Sidekick.Core/Update/UpdateManager.cs

```diff
@@ -0,0 +1,203 @@
+#else
+            _latestRelease = await GetLatestRelease();
+            if (_latestRelease != null)
+            {
+                var latestVersion = new Version(Regex.Match(_latestRelease.Tag, @"(\d+\.){2}\d+").ToString());
+                var currentVersion = AppDomain.CurrentDomain.GetAssemblies().FirstOrDefault(x => x.FullName.Contains("Sidekick")).GetName().Version;
+
+                var result = currentVersion.CompareTo(latestVersion);
+                return result < 0 ? true : false;
+            }
+                        
+            return false;
+#endif
+
+        }
+
+        /// <summary>
+        /// Trys to update sidekick
+        /// </summary>
+        /// <returns></returns>
+        public async Task<bool> UpdateSidekick()
+        {
+            if (await DownloadNewestRelease())
+            {
+                if (BackupFiles())
+                {
+                    ApplyUpdate();
+                    return true;
+                }
+            }
+
+            RollbackUpdate();
+            return false;
+        }
+
+        /// <summary>
+        /// Determines latest release on github. Pre-releases do not count as release, therefore we need to get the list of releases first, if no actual latest release can be found
+        /// </summary>
+        /// <returns></returns>
+        private async Task<GithubRelease> GetLatestRelease()
+        {
+            GithubRelease latestRelease = null;
+            try
+            {
+                using (var httpClient = new HttpClient())
+                {
+                    httpClient.BaseAddress = new Uri("https://api.github.com");
+                    httpClient.DefaultRequestHeaders.UserAgent.TryParseAdd("request");
+                    httpClient.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+
+                    var jsonOptions = new JsonSerializerOptions { IgnoreNullValues = true, PropertyNameCaseInsensitive = true };
+
+                    var response = await httpClient.GetAsync("/repos/domialex/Sidekick/releases/latest");
+                    if (response.IsSuccessStatusCode)
+                    {
+                        latestRelease = await JsonSerializer.DeserializeAsync<GithubRelease>(await response.Content.ReadAsStreamAsync(), jsonOptions);
+                    }
+                    else
+                    {
+                        //Get List of releases if there is no correct latest release ( should only happen if there are only pre-releases)
+                        var listResponse = await httpClient.GetAsync("/repos/domialex/Sidekick/releases");
+                        if (listResponse.IsSuccessStatusCode)
+                        {
+                            var githubReleaseList = await JsonSerializer.DeserializeAsync<GithubRelease[]>(await listResponse.Content.ReadAsStreamAsync(), jsonOptions);
+                            latestRelease = githubReleaseList[0];
```

Comment:
Should compare dates, the order might not be 100% reliable.

Reply:
It’s probably better to just get the release with the highest version number, rather than comparing dates, I guess?

Reply:
Even better!

Reply:
Done